src/main/java/apoc/algo/Similarity.java

```diff
@@ -0,0 +1,51 @@
+package apoc.algo;
+
+import apoc.util.Util;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Relationship;
+import org.neo4j.procedure.Context;
+import org.neo4j.procedure.Description;
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.UserFunction;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class Similarity {
+    @Context
+    public GraphDatabaseService db;
+
+    @UserFunction
+    @Description("apoc.algo.cosineSimilarity([[rel1A, rel1B], [rel2A, rel2B], …], weightPropertyName, defaultWeight: 0) " +
+            "given a collection of collected pairs of relationships, the weight property of the relationships, " +
+            "and a default weight (if relationship or property is null), calculate cosine similarity")
+    public Double cosineSimilarity(
+            @Name("relPairs") List<List<Relationship>> relPairs,
+            @Name("weightPropertyName") String weightPropertyName,
+            @Name(value = "defaultWeight", defaultValue = "0.0") Double defaultWeight) {
+
+        List<Double> weight1 = new ArrayList<>();
```

Comment:
Sensible suggestions, I made the changes accordingly. Please let me know if you see something I missed.